explorer/ui/app/root.tsx

```diff
@@ -1,17 +1,42 @@
-import { cssBundleHref } from '@remix-run/css-bundle'
-import type { LinksFunction } from '@remix-run/node'
+import type { LinksFunction, SerializeFrom } from '@remix-run/node'
 import stylesheet from '~/tailwind.css'
-import { Links, LiveReload, Meta, Outlet, Scripts, ScrollRestoration } from '@remix-run/react'
+import {
+  Links,
+  LiveReload,
+  Meta,
+  Outlet,
+  Scripts,
+  ScrollRestoration,
+  json,
+  useLoaderData,
+} from '@remix-run/react'
 import { Client, Provider, cacheExchange, fetchExchange } from 'urql'
+import { useEnv } from './lib/use-env'
 
 export const links: LinksFunction = () => [{ rel: 'stylesheet', href: stylesheet }]
+export type LoaderData = SerializeFrom<typeof loader>
 
 const client = new Client({
-  url: 'http://localhost:8080/query',
+  url: 'http://localhost:21335/query',
```

Comment:
Ideally, this URL should come from a config.

Reply:
I am trying to set it in the environment variables, not working, just pushing this out so that if the contractors run it locally they can use it with live data.